### PR TITLE
Report errors on SimpleRenderer on missing getters

### LINKED ISSUE #2667: 'simple' resolvers should somehow warn in mustachio if a parent node resolves for them
Somehow, it would be nice if a simple resolver would warn if it _could have_ resolved something if it was part of the visibleTypes, but failed to do so.  An example case that made me scratch my head for an hour or two was:

https://github.com/dart-lang/dartdoc/blob/09226603928907ddebb11054525b7690d021e701/lib/templates/html/class.html#L19

I had erroneously thought my refactor in #2665 meant I could delete `DefinedElementType` from the visibleTypes list.  When I did so and mustachio went to resolve `linkedName` here, it skipped over the `DefinedElementType` inside `publicSuperChainReversed` and instead resolved `linkedName` against the class.   This meant that the class's name repeated in the superchain list.

This was particularly troublesome to debug as unless someone was actually looking at the generated HTML there was no way to know, and everything looked correct in the debugger, the superchain list objects were correct, and so forth.  In fact, if it wasn't for that I remembered I had removed that and noticed that the superchain was a collection of `DefinedElementType`s I might still be scratching my head.

I am not sure how to make this warning happen without making simple resolvers much less simple...

### PR DESCRIPTION
**Pull request by srawlins**
Fixes #2667

If Mustachio is resolving a key and it finds a SimpleResolver (a resolver for a type not visible to Mustachio), and the key _is a getter_ on that type, then Mustachio will throw an error rather than continue to other types up the stack.

### CODE REVIEW

File: lib/src/mustachio/renderer_base.dart [hunk 1]
```diff
@@ -183,10 +191,21 @@ abstract class RendererBase<T> {
...
+            'a property on $T'));
       }
-    } else if (parent != null) {
+    }
+
+    if (this is SimpleRenderer && _invisibleGetters.contains(names.first)) {
+      var type = context.runtimeType;
+      throw MustachioResolutionError(node.keySpan.message(
+          '$firstName is a getter on $type, which is not visible to Mustache. '
+          'To render $firstName on $type, make it visible to mustache; to '
```
Comment (jcollins-g) [inline lib/src/mustachio/renderer_base.dart:203]: might be more specific to help with maintenance, e.g. `add to visibleTypes`

File: lib/src/mustachio/renderer_base.dart [hunk 2]
```diff
@@ -183,10 +191,21 @@ abstract class RendererBase<T> {
...
-            'to a property on $T'));
+            "chain: $contextChainString, after first resolving '$firstName' to "
+            'a property on $T'));
       }
-    } else if (parent != null) {
+    }
+
+    if (this is SimpleRenderer && _invisibleGetters.contains(names.first)) {
+      var type = context.runtimeType;
+      throw MustachioResolutionError(node.keySpan.message(
```
Comment (jcollins-g) [inline lib/src/mustachio/renderer_base.dart:201]: optional:  This seems like it should work, but there is no end to end test?

**Review by jcollins-g — APPROVED** [review]
Nice!

File: lib/src/mustachio/renderer_base.dart (hunk 2 above)
Comment (srawlins) [reply lib/src/mustachio/renderer_base.dart]: OK, there were definitely bugs :P  added two tests after fixing the bugs.

File: lib/src/mustachio/renderer_base.dart (hunk 1 above)
Comment (srawlins) [reply lib/src/mustachio/renderer_base.dart]: done

**Pull request merged**